projects/gateway2/translator/plugins/listeneroptions/query/query.go

```diff
@@ -73,22 +80,38 @@ func (r *listenerOptionQueries) GetAttachedListenerOptions(
 		return nil, err
 	}
 
-	if len(list.Items) == 0 {
+	listListenerSet := &solokubev1.ListenerOptionList{}
+	if parentListenerSet != nil {
+		nnListenerSet := types.NamespacedName{
+			Namespace: parentListenerSet.GetNamespace(),
+			Name:      parentListenerSet.GetName(),
+		}
+		if err := r.c.List(
+			ctx,
+			listListenerSet,
+			client.MatchingFieldsSelector{Selector: fields.OneTermEqualSelector(ListenerOptionTargetField, nnListenerSet.String())},
```

Comment:
Could this cause an issue if both the gw and the listenerset have the same name ?

Reply:
Possibly. I'll investigate.

Reply:
Updated the tests to confirm that there is no issue

Reply:
(and other tests confirmed that there IS an issue)